### ISSUE TITLE
Submit app to F-Droid

### ISSUE THREAD

**Issue by sbstn87** [wrwrabbit/Partisan-SMS#34]
Hi,

would you be willing to submit the app to F-Droid?

**Comment by BalooRJ1** [comment]
Seconding this. Came across this app from a Reddit thread. I've been using QKSMS and it hasnt been updated in quite a while.

**Comment by timonoj** [comment]
Yeah, there's interest in your app, but it would need to be integrated into the F-Droid builds. Thanks!
https://f-droid.org/en/contribute/

**Comment by wrwrabbit** [comment]
> Hi,
> 
> would you be willing to submit the app to F-Droid?

Sure. We'll work on it

**Comment by IzzySoft** [comment]
Can you please get your tag names, `versionCode` and `versionName` fixed? Currently, at v2.1.2 as well as v2.1.4, the APK identifies as

    package: name='org.cpartisans.psms' versionCode='2218' versionName='3.9.4'

`versionCode` is what Android uses internally to keep versions apart (and decide whether some APK is an update). Without that fixed, F-Droid could never update. `versionName` is for the user to be displayed – and would be quite confusing if never changed :wink:

I also wonder where you set this, as https://github.com/wrwrabbit/Partisan-SMS/blob/master/android-smsmms/build.gradle has it on 1.0+1 (which is wrong as well)…

**Comment by vivabelarus** [comment]
> Can you please get your tag names, `versionCode` and `versionName` fixed? Currently, at v2.1.2 as well as v2.1.4, the APK identifies as

Yes, we'll fix it. Initially, the idea was to pretend that this was not a Guerrilla SMS, but a QKSMS. This would allow a person, in the event of a police search, to say that this is a regular SMS application, it cannot send any encrypted SMS. We do this in our other project, Partisan Telegram. Later it became clear that in the case of the Partisan SMS, this does not make much sense. Not many people use QKSMS and therefore the presence of such an application in a political activist could indicate that it allows you to encrypt SMS. Therefore, we decided to give the application a separate icon and name. But the version remained from the original application :)

> I also wonder where you set this, as https://github.com/wrwrabbit/Partisan-SMS/blob/master/android-smsmms/build.gradle has it on 1.0+1 (which is wrong as well)…

This is left over from QKSMS. We decided not to fix what is already working

**Comment by IzzySoft** [comment]
Thanks for the explanation, @vivabelarus! Can you give me a ping please when tag names are matching `versionName`? I was asked to include your app with my F-Droid repo (until it makes it to F-Droid.org) and would need that for auto-updates.

**Comment by Shadeclan** [comment]
> > Can you please get your tag names, `versionCode` and `versionName` fixed? Currently, at v2.1.2 as well as v2.1.4, the APK identifies as
> 
> Yes, we'll fix it. Initially, the idea was to pretend that this was not a Partisan SMS, but a QKSMS. This would allow a person, in the event of a police search, to say that this is a regular SMS application, it cannot send any encrypted SMS. We do this in our other project, [Partisan Telegram](https://github.com/wrwrabbit/Partisan-Telegram-Android). Later it became clear that in the case of the Partisan SMS, this does not make much sense. Not many people use QKSMS and therefore the presence of such an application in a political activist could indicate that it allows you to encrypt SMS. Therefore, we decided to give the application a separate icon and name. But the version remained from the original application :)
> 
> > I also wonder where you set this, as https://github.com/wrwrabbit/Partisan-SMS/blob/master/android-smsmms/build.gradle has it on 1.0+1 (which is wrong as well)…
> 
> This is left over from QKSMS. We decided not to fix what is already working

I think there will be a lot more interest in your app now that Signal is getting rid of SMS support, at least here in the states and in parts of Europe where SMS is cheap.

**Comment by vivabelarus** [comment]
We are preparing an app update and will notify you when we are done, as promised.

**Comment by netllama** [comment]
Do you have a rough ETA when that will happen?  I understand this work takes time, but its been nearly 4 months.

**Comment by IzzySoft** [comment]
I'm just looking to clean up my issue tracker, where a request on including your app is pending for half a year now, waiting for the "proper release" here. So an ETA would be much appreciated. If it's pending indefinitely, I'd rather have to close the inclusion request as idle/inactive.

**Comment by vivabelarus** [comment]
Hello. Sorry for the long wait. An update is planned for this month, in April. Unless there are unforeseen circumstances.

**Comment by Dyras** [comment]
Any news?

**Comment by vivabelarus** [comment]
We have finally released an update: Fixed in 4.0.0.
https://github.com/wrwrabbit/Partisan-SMS/releases/tag/v4.0.0

@IzzySoft, now tag name matches versionName. Sorry for such a huge delay.

**Comment by IzzySoft** [comment]
Thanks, that looks good so far! I just started setting it up for my repo, but miss quite some data: there's no icon, and there are no screenshots. Could you provide an icon and 2 (or more) screenshots? You know, first impression and all that.

Oh, and further: the URL next to the project's short description goes to the QKSMS page on PlayStore: https://qklabs.com/download :see_no_evil:

**Comment by vivabelarus** [comment]
@IzzySoft, are these images suitable? [PSMS.zip](https://github.com/wrwrabbit/Partisan-SMS/files/11650319/PSMS.zip)

> Oh, and further: the URL next to the project's short description goes to the QKSMS page on PlayStore: https://qklabs.com/download 🙈

Oh thanks!

**Comment by Dyras** [comment]
@vivabelarus 
Any chance you could write some more details about how the app works? There is little to no info about how the encryption works, which version of QKSMS is it based on, which features did you add?

**Comment by IzzySoft** [comment]
@vivabelarus thanks for fixing that URL :smile: Will you give me another ping when the other issues have been addressed (i.e. when an icon and some screenshots are available)? Some of the details @Dyras just requested would be a good fit in the app description, too (the Readme doesn't show much details).

**Comment by IzzySoft** [comment]
My [corresponding issue](https://gitlab.com/IzzyOnDroid/repo/-/issues/306) just pinged me (due-date reached) – so may I ask where we stand here?

**Comment by IzzySoft** [comment]
Shifting the due date once more. But if we cannot progress, I'll have to close it on the next round.

**Comment by Dyras** [comment]
I think it's safe to say that this app is dead, unfortunately. No response in 4 months isn't exactly a good sign for a security-focused app.

**Comment by IzzySoft** [comment]
> No response in 4 months isn't exactly a good sign for a security-focused app.

I certainly agree with that – and closed the inclusion request in my repo's tracker 2 weeks ago for that reason. Sad, we've been so close already :cry:
